### PR TITLE
Test/systemctl segfault

### PR DESCRIPTION
**Pull request by schrodit**
**What this PR does / why we need it**:
Added a integration that checks if `journalctl` on a node contains any `segfaults`.
It also verifies that `systemctl` is working on the node.

**Special notes for your reviewer**:

cc @marwinski @gehoern

Is the load really needed as in your demo today it was also during kubelet startup?
So would it be enough to just check journalctl and systemctl?

**Release note**:

```improvement operator
Added a test to validate if systemctl on the node's operating system runs without errors.
```

### CODE REVIEW

File: test/framework/k8s_utils.go [hunk 1]
```diff
@@ -219,6 +221,88 @@ func DeleteResource(ctx context.Context, k8sClient kubernetes.Interface, resourc
...
+				return retry.Ok()
+			}
+			return retry.MinorError(err)
+		}
+		return retry.MinorError(errors.New("Object still exists"))
+	})
+}
+
+// ScaleDeployment scales a deployment and waits until it is scaled
+func ScaleDeployment(setupContextTimeout time.Duration, client client.Client, desiredReplicas *int32, name, namespace string) (*int32, error) {
```
Comment (rfranzke) [inline test/framework/k8s_utils.go]: Why these functions are duplicated here? We already have them e.g. [here](https://github.com/gardener/gardener/blob/f2f95d9fa3690cbb0d2a1db6c04093728cbcaec8/extensions/test/integration/operation/shoot_operation.go#L65-L93). Can you make them re-usable instead of duplicating?

File: test/framework/k8s_utils.go [hunk 2]
```diff
@@ -383,3 +488,8 @@ func DeployRootPod(ctx context.Context, c client.Client, namespace string, noden
 	}
 	return &rootPod, nil
 }
+
+// DeployRootPod deploys a pod with root permissions for testing purposes.
+func (*CommonFramework) DeployRootPod(ctx context.Context, c client.Client, namespace string, nodename *string) (*corev1.Pod, error) {
```
Comment (rfranzke) [inline test/framework/k8s_utils.go]: What is the purpose of this wrapper funciton?

File: test/framework/k8s_utils.go [hunk 3]
```diff
@@ -316,6 +400,27 @@ func NewClientFromServiceAccount(ctx context.Context, k8sClient kubernetes.Inter
...
+			log.Infof("Waiting for %s to be ready!!", podName)
+			return retry.MinorError(fmt.Errorf(`pod "%s/%s" is not ready: %v`, podNamespace, podName, err))
+		}
+
+		return retry.Ok()
+	})
+}
+
+// WaitUntilPodIsRunning waits until the pod with <podName> is running
+func (f *CommonFramework) WaitUntilPodIsRunning(ctx context.Context, podName, podNamespace string, c kubernetes.Interface) error {
```
Comment (rfranzke) [inline test/framework/k8s_utils.go]: Same here, what is the purpose of this wrapper function?

File: test/framework/k8s_utils.go (hunk 1 above)
Comment (schrodit) [reply test/framework/k8s_utils.go]: This is currently duplicated because we need have the old framework in `test/integration/framework` until the last usage in the extension networking tests are removed.
I wanted to update the networking tests next week and then remove the old framework after the next releases.
Until now a marked the package as deprecated.

File: test/framework/k8s_utils.go (hunk 2 above)
Comment (schrodit) [reply test/framework/k8s_utils.go]: removed

File: test/framework/dump.go [hunk 4]
```diff
@@ -45,28 +46,37 @@ func (f *CommonFramework) dumpDefaultResourcesInAllNamespaces(ctx context.Contex
...
-			result = multierror.Append(result, fmt.Errorf("unable to fetch information of Services from namespace %s: %s", ns.Name, err.Error()))
+		if err := f.DumpDefaultResourcesInNamespace(ctx, ctxIdentifier, k8sClient, ns.Name); err != nil {
+			result = multierror.Append(result, err)
 		}
 	}
 	return result
 }
 
+// DumpDefaultResourcesInNamespace dumps all default K8s resources of a namespace.
+func (f *CommonFramework) DumpDefaultResourcesInNamespace(ctx context.Context, ctxIdentifier string, k8sClient kubernetes.Interface, namespace string) error {
```
Comment (danielfoehrKn) [inline test/framework/dump.go]: Is that related to the Testcase introduced in the PR? It seem to me like this is a change to the CommonFramwork instead?

File: test/suites/shoot/run_suite_test.go [hunk 5]
```diff
@@ -17,9 +17,10 @@ package shoot_suite_test
 import (
 	"flag"
 	"fmt"
-	"github.com/gardener/gardener/test/framework/config"
 	"os"
 
+	"github.com/gardener/gardener/test/framework/config"
```
Comment (danielfoehrKn) [inline test/suites/shoot/run_suite_test.go]: sort imports

Comment (schrodit) [reply test/suites/shoot/run_suite_test.go]: done

File: test/framework/dump.go (hunk 4 above)
Comment (schrodit) [reply test/framework/dump.go]: you're right. I split the PR and created a new one just for the other changes

File: test/integration/shoots/operations/clusterautoscaler.go [hunk 6]
```diff
@@ -31,6 +31,7 @@ package operations
 
 import (
 	"context"
+	"github.com/gardener/gardener/test/framework/resources/templates"
```
Comment (timebertt) [inline test/integration/shoots/operations/clusterautoscaler.go]: sort imports

File: test/framework/rootpod_executor.go [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+// RunCommand executes a command on the node the root pod is running
+func (e *RootPodExecutor) RunCommand(ctx context.Context, command string) ([]byte, error) {
+	if e.Pod == nil {
+		if err := e.Deploy(ctx); err != nil {
+			return nil, err
+		}
+	}
+
+	command = fmt.Sprintf("chroot /hostroot %s", command)
+	reader, err := kubernetes.NewPodExecutor(e.client.RESTConfig()).Execute(ctx, e.Pod.Namespace, e.Pod.Name, e.Pod.Spec.Containers[0].Name, command)
```
Comment (timebertt) [inline test/framework/rootpod_executor.go]: This seems kind of fishy, to hand the `RESTConfig` of a client to the executor, which then creates a new client.
If I see it correctly `kubernetes.NewPodExecutor()` is only used in `/test`, so can you move it to `test/framework` and refactor it a bit to reuse an existing client, please?

File: test/integration/shoots/logging/seed_log.go [hunk 8]
```diff
@@ -17,6 +17,7 @@ package logging
 import (
 	"context"
 	"fmt"
+	"github.com/gardener/gardener/test/framework/resources/templates"
```
Comment (timebertt) [inline test/integration/shoots/logging/seed_log.go]: sort imports

File: test/integration/shoots/operatingsystem/os.go [hunk 9]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```
Comment (timebertt) [inline test/integration/shoots/operatingsystem/os.go]:
```suggestion
// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

File: test/integration/shoots/operatingsystem/os.go [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+		- No segfault in the journalctl logs
+
+ **/
+
+package operatingsystem
+
+import (
+	"context"
+	"github.com/gardener/gardener/test/framework"
+	"github.com/gardener/gardener/test/framework/resources/templates"
```
Comment (timebertt) [inline test/integration/shoots/operatingsystem/os.go:32]: sort imports

**Review by timebertt — COMMENTED** [review]
Only some nits

**Review by dguendisch — DISMISSED** [review]

File: test/framework/rootpod_executor.go (hunk 7 above)
Comment (schrodit) [reply test/framework/rootpod_executor.go]: nice catch. Refactored now to be in the framework. PTAL

**Review by timebertt — APPROVED** [review]
Thanks, very nice!
/lgtm now
